packages/containerd/3001-cri-set-default-RLIMIT_NOFILE.patch

```diff
@@ -0,0 +1,87 @@
+From 2e34c4dd462f0cdb05e5cea55a8618f9ce443499 Mon Sep 17 00:00:00 2001
+From: Zac Mrowicki <mrowicki@amazon.com>
+Date: Thu, 22 Oct 2020 20:44:38 +0000
+Subject: [PATCH] cri: set a default RLIMIT_NOFILE of 1048576
+
+The `cri` plugin currently inherits the limit from the default OCI spec
+or the containerd process.  This change sets the default RLIMIT_NOFILE
+to 1048576 in the OCI spec for any container spawned using `cri`.
+---
+ vendor/github.com/containerd/cri/pkg/config/config.go |  8 ++++++++
+ .../containerd/cri/pkg/containerd/opts/spec.go        | 11 +++++++++++
+ .../containerd/cri/pkg/server/container_create.go     | 11 +++++++++++
+ 3 files changed, 30 insertions(+)
+
+diff --git a/vendor/github.com/containerd/cri/pkg/config/config.go b/vendor/github.com/containerd/cri/pkg/config/config.go
+index 7c5f9eb..f7c3f28 100644
+--- a/vendor/github.com/containerd/cri/pkg/config/config.go
++++ b/vendor/github.com/containerd/cri/pkg/config/config.go
+@@ -204,6 +204,12 @@ type PluginConfig struct {
+ 	// DisableProcMount disables Kubernetes ProcMount support. This MUST be set to `true`
+ 	// when using containerd with Kubernetes <=1.11.
+ 	DisableProcMount bool `toml:"disable_proc_mount" json:"disableProcMount"`
++        // ProcessRLimitNoFileSoft sets the soft limit of maximum file
++        // descriptors each container process can use.
++        ProcessRLimitNoFileSoft int `toml:"process_rlimit_no_file_soft" json:"process_rlimit_no_file_soft"`
++        // ProcessRLimitNoFileHard sets the hard limit of maximum file
++        // descriptors each container process can use.
++        ProcessRLimitNoFileHard int `toml:"process_rlimit_no_file_hard" json:"process_rlimit_no_file_hard"`
+ }
+ 
+ // X509KeyPairStreaming contains the x509 configuration for streaming
+@@ -271,6 +277,8 @@ func DefaultConfig() PluginConfig {
+ 		},
+ 		MaxConcurrentDownloads: 3,
+ 		DisableProcMount:       false,
++                ProcessRLimitNoFileSoft:    65536,
++                ProcessRLimitNoFileHard:    1048576,
+ 	}
+ }
+ 
```

Comment:
The existing code is indented with tabs, but you've used spaces. It'd be good to match the whitespace convention in case this is headed upstream in some form.

Reply:
Good catch!

Reply:
For Go changes, it's a good idea to run them through `go fmt` and `goimports`.  This will fix issues like tabs vs spaces.